src/fsharp/CompilerConfig.fs

```diff
@@ -829,6 +832,15 @@ type TcConfigBuilder =
     member tcConfigB.AddPathMapping (oldPrefix, newPrefix) =
         tcConfigB.pathMap <- tcConfigB.pathMap |> PathMap.addMapping oldPrefix newPrefix
 
+    member tcConfigB.FileSystem =
+                if tcConfigB.filesystem = Unchecked.defaultof<IFileSystem> then
+                    lock tcConfigB (fun () ->
+                        if tcConfigB.filesystem = Unchecked.defaultof<IFileSystem> then
+                            tcConfigB.filesystem <- DefaultFileSystem() :> IFileSystem
```

Comment:
Shouldn't the file system be one per checker? What's the planned approach to override it?
It would be nice to make `DefaultFileSystem` members virtual, so clients don't have to make boilerplate implementations.

Reply:
> It would be nice to make DefaultFileSystem members virtual, so clients don't have to make boilerplate implementations.

Yeah, that's what I've done (didn't push that change yet).